### DISCUSSION TITLE
Would it be possible to make a fully fledged 3rd party client for Linux with this?

### DISCUSSION THREAD

**Discussion by nyabinary** [tidal-music/discussions#7] [General]
Linux desperately needs a native client for Tidal.

**Comment by twnlink** [comment]
No, the TIDAL api does not support playback.

**Comment by stefanmoro** [comment]
Hey @Nyabinary and welcome to TIDAL community discussions.

As already pointed out, TIDAL Developer Platform is in early stages and as of today it lacks all the functionality needed to build a fully fledged 3rd party client. We will continue to open up functionality as soon as we have something new, and, hopefully, we will one day reach a state where such a client can be built.

Thanks.

**Comment by tehkillerbee** [comment]
> Linux desperately needs a native client for Tidal.

Right now playback through mopidy is possible through open source efforts. It would be very exciting to to integrate this new API as the backend.

**Reply by stefanmoro** [reply]
Hi @tehkillerbee.

The mopidy integration, and the python-tidal lib that it depends on, represents an unauthorized use of TIDAL’s internal API. Our hope is that we will be able to supply the functionality needed for such use cases via the new TIDAL Developer Platform in the future.

**Reply by nyabinary** [reply]
@stefanmoro Well you better develop and release your own official TIDAL client for Linux or open up the API, pick your poison.

**Reply by tehkillerbee** [reply]
> The mopidy integration, and the python-tidal lib that it depends on, represents an unauthorized use of TIDAL’s internal API. Our hope is that we will be able to supply the functionality needed for such use cases via the new TIDAL Developer Platform in the future.

@stefanmoro You are very welcome to suggest any ways that python-tidal and the derivatives can align better with TIDALs ideas of an open API. Feel free to get in touch if you have good ideas on how we can achieve this.

I know a lot of previous Spotify users that migrated to Tidal, simply because a (good) API was available, even though it was only made possible through reverse engineering efforts by the community.

**Reply by stefanmoro** [reply]
With TIDAL Developer Platform we are taking the first steps towards a more open platform, and our hope is that we will be able to supply more functionality going forward. Throughout this journey, it is important for us to work closely together with labels, artists and content owners to make sure the solutions we build are beneficial for all involved parties.

We understand that we currently do not support all use cases for all of our customers, but this is something we are continuously working towards improving. In addition to our continuous work to expand our product offering, we are now also taking the first steps to opening up our platform for others to build upon, and we hope to open up more functionality soon.

**Reply by sachinsenal0x64** [reply]
@tehkillerbee That's right. I prefer Tidal because of its api for this reason. :)

**Reply by giggywithit** [reply]
I make immaculate all-in-one audiophile amplifiers that use the Raspberry Pi platform and use Raspotify flawlessly. The model that I made for my mother has been running for 2 years. The only reason that I use Spotify is it's integration and the impression that it would soon have a lossless format to use. Lossless on Spotify hasn't materialized and I'm not one to wait around more than necessary.... I would be glad to install software and use Tidal exclusively if and when possible. MQA isn't a requisite, although my DAC does support these files, FLAC is fine. Since reliability and flawless operation are my standard, Tidal functionality will need to be at least the same level that I get from the Raspotify community. The only thing missing is better sound quality that it is designed to acheive

**Comment by nyabinary** [comment]
@stefanmoro any update to this/
